astrogrism/simulate/simulate.py

```diff
@@ -0,0 +1,84 @@
+from os import environ
+from pathlib import Path
+from tarfile import open as tar_open
+from tempfile import mkdtemp
+from warnings import warn
+
+from astropy.utils.data import download_file
+
+
+def generate_simulation_spectrum(grism, detector=None, temp_path=mkdtemp()):
+    """
+    Initializes and uses STSynphot to generate a Vega spectrum within the bandpass of a given grism
+
+    Parameters
+    ----------
+    grism : str
+        String representation of one of the four supported HST Grisms
+        Valid grisms: G141, G102, G280, G800L
+
+    detector : int
+        For detectors with multiple chips, specifies which chip to simulate
+        Only useful for G280 and G800L Grisms
+
+    temp_path : str
+        Path to download necessary files for STSynphot. Fallsback to Python's
+        default temporary folder location
+
+    """
+    SIM_DATA_DIR = Path(temp_path) / "astrogrism_simulation_files"
+
+    if detector not in (1, 2, None):
+        raise ValueError("Invalid detector argument. Please choose 1 or 2")
+
+    # Prepare environment required for stsynphot to be imported
+    environ['PYSYN_CDBS'] = str(SIM_DATA_DIR / 'grp' / 'redcat' / 'trds')
+
+    # Always download our data files fresh. Ensures changes from redcat team will be picked up
+    # Download HST Instrument Data Files
+    SIM_DATA_DIR.mkdir(parents=True, exist_ok=True)
+    hst_data_files_archive = Path(download_file('https://ssb.stsci.edu/trds/tarfiles/synphot1.tar.gz')) # noqa
+    with tar_open(hst_data_files_archive) as tar:
+        tar.extractall(path=SIM_DATA_DIR)
+    # Download Vega CALSPEC Reference Atlas
+    calspec_dir = SIM_DATA_DIR / 'grp' / 'redcat' / 'trds' / 'calspec'
+    calspec_dir.mkdir(parents=True, exist_ok=True)
+    vega_reference_atlas = Path(download_file('https://archive.stsci.edu/hlsps/reference-atlases/cdbs/current_calspec/alpha_lyr_stis_010.fits')) # noqa
+    vega_reference_atlas.replace(calspec_dir / 'alpha_lyr_stis_010.fits')
```

Comment:
Not sure what you are using `vega_reference_atlas` for but if you only want the file name, I think you can get it from `stsynphot.Vega.meta['header']['filename']`.

https://github.com/spacetelescope/synphot_refactor/blob/e51d7605349eebea0921c7d72da674b0af7d6db9/synphot/spectrum.py#L1384

Reply:
So this is the weird thing... The point of this code block is to "prepare" the folder structure in advance of importing stsynphot (hence https://github.com/spacetelescope/astrogrism/pull/18/files#diff-a44ad3e831c16508917324e007e1c88a870570691f40e4410b6fd5782c37b6a1R51). So at this point, I actually don't have access to `stsynphot` to pull out the filename, if that makes sense? That line is to rename the random hash `astropy.download_file` gives the file

Reply:
I guess I still don't get it. I don't see this variable used below and I know `stsynphot` only cares about `PYSYN_CDBS`.

Reply:
Ah! I thought it was a string operation. I didn't realize `Path.replace` is like `shutil.move`. Ok, clear now. 😅

Reply:
Thinking about this more, maybe you want a copy instead of move. That way, if you have downloaded it before, it will grab the cache instead of hitting the Internet again. To use cache, you need to pass `cache=True` into `download_file`.

Reply:
So the reason why I always download from the internet again was the scenario you painted earlier, where redcat comes in and occasionally updates the reference files. If we cache the files, there could theoretically be a case where the cached version differs from the online version, right?

Reply:
I don't think RedCat ever overwrites new version with the same filename. If they ever release a new version of Vega, it would be something like `alpha_lyr_stis_011.fits` (incremented from `alpha_lyr_stis_010.fits`). So, if you hardcode the filename, it is always going to be the same. Does that make sense?

Reply:
Ahh, ok. In that case caching makes sense. Thanks!